addons/sourcemod/scripting/vsh/bosses/boss_uberranger.sp

```diff
@@ -0,0 +1,423 @@
+		for (int i = 0; i < sizeof(g_strUberRangerBackStabbed); i++) PrecacheSound(g_strUberRangerBackStabbed[i]);
+		for (int i = 0; i < sizeof(g_strUberRangerJump); i++) PrecacheSound(g_strUberRangerJump[i]);
+		
+		AddFileToDownloadsTable("materials/models/player/boss/uber_ranger/uberranger_backpack.vmt");
+		AddFileToDownloadsTable("materials/models/player/boss/uber_ranger/uberranger_backpack.vtf");
+		AddFileToDownloadsTable("materials/models/player/boss/uber_ranger/uberranger_body.vmt");
+		AddFileToDownloadsTable("materials/models/player/boss/uber_ranger/uberranger_body.vtf");
+		AddFileToDownloadsTable("materials/models/player/boss/uber_ranger/uberranger_head.vmt");
+		AddFileToDownloadsTable("materials/models/player/boss/uber_ranger/uberranger_head.vtf");
+		
+		AddFileToDownloadsTable("models/player/vsh_rewrite/uber_ranger/uber_ranger.mdl");
+		AddFileToDownloadsTable("models/player/vsh_rewrite/uber_ranger/uber_ranger.sw.vtx");
+		AddFileToDownloadsTable("models/player/vsh_rewrite/uber_ranger/uber_ranger.vvd");
+		AddFileToDownloadsTable("models/player/vsh_rewrite/uber_ranger/uber_ranger.phy");
+		AddFileToDownloadsTable("models/player/vsh_rewrite/uber_ranger/uber_ranger.dx80.vtx");
+		AddFileToDownloadsTable("models/player/vsh_rewrite/uber_ranger/uber_ranger.dx90.vtx");
+	}
+	
+	public void Destroy()
+	{
+		SetEntityRenderColor(this.iClient, 255, 255, 255, 255);
+		
+		if (g_aUberRangerColorList != null)
+			delete g_aUberRangerColorList;
+	}
+};
+
+methodmap CMinionRanger < SaxtonHaleBase
+{
+	public CMinionRanger(CMinionRanger boss)
+	{
+		boss.iBaseHealth = 600;
+		boss.iHealthPerPlayer = 0;
+		boss.nClass = TFClass_Medic;
+		boss.iMaxRageDamage = -1;
+		boss.flWeighDownTimer = -1.0;
+		boss.bCanBeHealed = true;
+		boss.bMinion = true;
+		
+		EmitSoundToClient(boss.iClient, SOUND_ALERT);	//Alert player as he spawned
+	}
+	
+	public bool IsBossHidden()
+	{
+		return true;
+	}
+	
+	public void OnSpawn()
+	{
+		char sMedigunAttribs[128];
+		Format(sMedigunAttribs, sizeof(sMedigunAttribs), "7 ; 0.4");
+		this.CallFunction("CreateWeapon", 211, "tf_weapon_medigun", 10, TFQual_Collectors, sMedigunAttribs);
+		
+		/*
+		Medigun attributes:
+		
+		7: slower heal rate
+		*/
+		
+		char sSawAttribs[128];
+		Format(sSawAttribs, sizeof(sSawAttribs), "2 ; 1.25 ; 5 ; 1.2 ; 17 ; 0.25 ; 252 ; 0.5 ; 259 ; 1.0");
+		int iWeapon = this.CallFunction("CreateWeapon", 37, "tf_weapon_bonesaw", 10, TFQual_Collectors, sSawAttribs);
+		if (iWeapon > MaxClients)
+			SetEntPropEnt(this.iClient, Prop_Send, "m_hActiveWeapon", iWeapon);
+		
+		/*
+		Ubersaw attributes:
+		
+		2: damage bonus
+		5: slower firing speed
+		17: add uber on hit
+		252: reduction in push force taken from damage
+		259: Deals 3x falling damage to the player you land on
+		*/
+		
+		
+		//We're selecting their color from a preset list
+		
+		//Checking if the list is there at all
+		if (g_aUberRangerColorList.Length <= 0 || g_aUberRangerColorList == null)
```

Comment:
`g_aUberRangerColorList == null` need to be checked before `g_aUberRangerColorList.Length <= 0` (swap both checks). Otherwise `g_aUberRangerColorList.Length` would be checked first and throw null/invalid handle error